### PR TITLE
feat(ui): export generations async on cloud

### PR DESCRIPTION
**Pull request by marliessophie**

### CODE REVIEW

File: worker/src/features/batchExport/handleBatchExportJob.ts [hunk 1]
```diff
@@ -116,27 +158,63 @@ const getDatabaseReadStream = async ({
...
+      );
+    }
+    case "traces": {
+      const { orderByCondition, filterCondition, observationTimeseriesFilter } =
+        parseTraceAllFilters({
+          projectId,
+          orderBy,
+          filter: filter
+            ? [...filter, createdAtCutoffFilter]
+            : [createdAtCutoffFilter],
```
Comment (hassiebp) [inline worker/src/features/batchExport/handleBatchExportJob.ts:210]: nit: can be simply [...(filter || []), createdCutoffFilter]

File: web/src/components/table/use-cases/generations.tsx [hunk 2]
```diff
@@ -763,36 +768,44 @@ export default function GenerationsTable({
...
-                  onClick={() =>
-                    void handleExport(key as BatchExportFileFormat)
-                  }
-                >
-                  as {options.label}
-                </DropdownMenuItem>
-              ))}
-            </DropdownMenuContent>
-          </DropdownMenu>
+          plan && isCloudPlan(plan) ? (
```
Comment (hassiebp) [inline web/src/components/table/use-cases/generations.tsx:771]: question: this effectively disables the exports for self-hosters right? Is that intended?

File: packages/shared/src/server/queries/createTracesQuery.ts [hunk 3]
```diff
@@ -1,4 +1,50 @@
...
+      : Prisma.empty;
+
+  const searchCondition = input.searchQuery
+    ? Prisma.sql`AND (
+    t."id" ILIKE ${`%${input.searchQuery}%`} OR 
+    t."external_id" ILIKE ${`%${input.searchQuery}%`} OR 
+    t."user_id" ILIKE ${`%${input.searchQuery}%`} OR 
+    t."name" ILIKE ${`%${input.searchQuery}%`}
+  )`
+    : Prisma.empty;
```
[the remark this answers is not in the document]
Comment (marliessophie) [reply packages/shared/src/server/queries/createTracesQuery.ts:39]: Copied from somewhere else, browsed through docs, don't think we're exposed to vulnerability here, checked with Hassieb too.

[the remark this answers is not in the document]
Comment (marcklingen) [reply packages/shared/src/server/queries/createTracesQuery.ts:39]: Prisma.sql makes this safe

**Review by hassiebp — APPROVED** [review]

**Pull request merged**